functions.php

```diff
@@ -617,3 +617,14 @@ function ($theme_json) {
         return $theme_json->update_with($new_color_palette);
     }
 );
+
+// Maximum revisions to keep whenever the editor save a post
+add_filter(
+    'wp_revisions_to_keep',
+    function ($revisions) {
+        $revisions_to_keep = get_option('revisions_to_keep');
+        return $revisions_to_keep ?: $revisions;
+    },
+    10,
+    2
```

Comment:
As we are accepting 1 argument in function, we need to pass 1 instead of 2, 
Ref. https://developer.wordpress.org/reference/functions/add_filter/